include/server/mir/frontend/session.h

```diff
@@ -51,6 +51,9 @@ class Session
 public:
     virtual ~Session() = default;
 
+    /// Throws std::out_of_range if the surface is invalid
+    virtual SurfaceId get_surface_id(Surface* surface) const = 0;
```

Comment:
I think we'd typically spell this `virtual SurfaceId get_surface_id(std::weak_ptr<Surface> surface) const = 0`.

Generally I try to avoid raw pointers (at least in interfaces); either one of the various smart pointers, or a reference, tell you more about how thing will be used, what valid values are, and so on.

Reply:
I think raw pointers are fine. They (should) imply no transfer of ownership and therefore make no lifetime guarantees beyond the duration of the call.